src/App.tsx

```diff
@@ -1,61 +1,140 @@
+import { FC, ChangeEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserId(userId: number): User | null {
+  return (usersFromServer.find(user => (user.id === userId)) || null);
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserId(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { id, value } = event.target;
+
+    switch (id) {
+      case 'titleInputForm':
+        setIsTitleError(false);
+        setTodoTitle(value);
+        break;
+
+      case 'userInputForm':
+        setIsUserError(false);
+        setSelectedUser(getUserId(Number(value)));
+        break;
+
+      default:
+        throw new Error('error');
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setIsTitleError(true);
+    }
+
+    if (!selectedUser) {
+      setIsUserError(true);
+    }
+
+    if (!todoTitle || !selectedUser || todoTitle.trim() === '') {
```

Comment:
Since you have already checked for !todoTitle in line 51, you can remove the !todoTitle condition in this line.